File: skills/code-optimizer/references/memory/node/code.ts
export type Row = {
  id: number;
  payload: string;
};

export function summarizeBad(rows: Row[]): number {
  const copied = rows.map((row) => ({
    id: row.id,
    tokens: row.payload.split(",").map((part) => part.trim().toLowerCase()),
  }));
  const flattened = copied.flatMap((row) => row.tokens);
  const filtered = flattened.filter((token) => token.length > 3);
  return filtered.length;
}

export function summarizeGood(rows: Row[]): number {
  let total = 0;
  for (const row of rows) {
    for (const part of row.payload.split(",")) {
      const token = part.trim().toLowerCase();
      if (token.length > 3) {
        total += 1;
      }
    }
  }
  return total;
}

File: skills/code-optimizer/references/memory/python/code.py
from dataclasses import dataclass


@dataclass(frozen=True)
class Row:
  id: int
  payload: str


def summarize_bad(rows: list[Row]) -> int:
  copied = [
    {
      "id": row.id,
      "tokens": [part.strip().lower() for part in row.payload.split(",")],
    }
    for row in rows
  ]
  flattened = [token for row in copied for token in row["tokens"]]
  filtered = [token for token in flattened if len(token) > 3]
  return len(filtered)


def summarize_good(rows: list[Row]) -> int:
  total = 0
  for row in rows:
    for part in row.payload.split(","):
      token = part.strip().lower()
      if len(token) > 3:
        total += 1
  return total
